client/my-sites/site-settings/form-general.jsx

```diff
@@ -609,10 +617,33 @@ export class SiteSettingsFormGeneral extends Component {
 
 		const LaunchCard = showPreviewLink ? CompactCard : Card;
 
+		if ( isSiteOnECommerceTrial ) {
+			const eCommerceTrialUpsellText = translate(
+				'Before you can share your store with the world, you need to {{NoticeAction}}pick a plan{{/NoticeAction}}.',
+				{
+					components: {
+						NoticeAction: (
+							<NoticeAction href={ `/plans/${ siteSlug }?plan=ecommerce-bundle-monthly` } />
+						),
+					},
+				}
+			);
+			eCommerceTrialUpsell = (
+				<Notice
+					className="site-settings__ecommerce-trial-notice"
+					icon="info"
+					showDismiss={ false }
+					text={ eCommerceTrialUpsellText }
+					isCompact={ false }
+				></Notice>
```

Comment:
Minor stylistic note: You can also use `/>` to close off the element.
```suggestion
				/>
```